test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,352 @@
+import deploymentData from "../json/safeDeployment.json";
+import safeRuntimeBytecode from "../json/safeRuntimeBytecode.json";
+import { executeContractCallWithSigners } from "../../src/utils/execution";
+
+const SAFE_SINGLETON_ADDRESS = "0x477C3fb2D564349E2F95a2EF1091bF9657b26145";
+
+const SAFE_SINGLETON_L2_ADDRESS = "0x551A2F9a71bF88cDBef3CBe60E95722f38eE0eAA";
+
+const COMPATIBILITY_FALLBACK_HANDLER = "0x4c95c836D31d329d80d696cb679f3dEa028Ad4e5";
+
+const FALLBACK_HANDLER_STORAGE_SLOT = "0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5";
+
+const GUARD_STORAGE_SLOT = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
+
+const migrationPaths = [
+    {
+        testSuiteName: "1.3.0 to 1.5.0",
+        from: { safeDeploymentData: deploymentData.safe130, safeL2DeploymentData: deploymentData.safe130l2 },
+        to: {
+            safeRuntimeBytecode: safeRuntimeBytecode.safe150,
+            safeL2RuntimeBytecode: safeRuntimeBytecode.safe150l2,
+            fallbackHandlerRuntimeBytecode: safeRuntimeBytecode.safe150CompatibilityFallbackHandler,
+        },
+    },
+    {
+        testSuiteName: "1.3.0 to 1.4.1",
+        from: { safeDeploymentData: deploymentData.safe130, safeL2DeploymentData: deploymentData.safe130l2 },
+        to: {
+            safeRuntimeBytecode: safeRuntimeBytecode.safe141,
+            safeL2RuntimeBytecode: safeRuntimeBytecode.safe141l2,
+            fallbackHandlerRuntimeBytecode: safeRuntimeBytecode.safe141fallbackHandler,
+        },
+    },
+    {
+        testSuiteName: "1.4.1 to 1.5.0",
+        from: { safeDeploymentData: deploymentData.safe141, safeL2DeploymentData: deploymentData.safe141l2 },
+        to: {
+            safeRuntimeBytecode: safeRuntimeBytecode.safe150,
+            safeL2RuntimeBytecode: safeRuntimeBytecode.safe150l2,
+            fallbackHandlerRuntimeBytecode: safeRuntimeBytecode.safe150CompatibilityFallbackHandler,
+        },
+    },
+];
+
+describe("SafeMigration Library", () => {
+    const migratedInterface = new ethers.Interface(["function masterCopy() view returns(address)"]);
+
+    migrationPaths.forEach(({ testSuiteName, from, to }) => {
+        describe(testSuiteName, () => {
+            const setupTests = deployments.createFixture(async ({ deployments }) => {
+                await deployments.fixture();
+                const signers = await ethers.getSigners();
+                const [user1] = signers;
+
+                // Set the runtime code for hardcoded addresses, so the expected events are emitted
+                await hre.network.provider.send("hardhat_setCode", [SAFE_SINGLETON_ADDRESS, to.safeRuntimeBytecode]);
+                await hre.network.provider.send("hardhat_setCode", [SAFE_SINGLETON_L2_ADDRESS, to.safeL2RuntimeBytecode]);
+                await hre.network.provider.send("hardhat_setCode", [COMPATIBILITY_FALLBACK_HANDLER, to.fallbackHandlerRuntimeBytecode]);
```

Comment:
if you add the contract to the deployment scripts you'll be able to get rid of this code